airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/DetectStreamToFlush.java

```diff
@@ -210,11 +211,18 @@ long estimateSizeOfRunningWorkers(final StreamDescriptor stream, final long curr
    */
   @VisibleForTesting
   List<StreamDescriptor> orderStreamsByPriority(final Set<StreamDescriptor> streams) {
+    // eagerly pull attributes so that values are consistent throughout comparison
+    final Map<StreamDescriptor, Optional<Long>> sdToQueueSize = streams.stream()
+        .collect(Collectors.toMap(s -> s, bufferDequeue::getQueueSizeBytes));
+
+    final Map<StreamDescriptor, Optional<Instant>> sdToTimeOfLastRecord = streams.stream()
+        .collect(Collectors.toMap(s -> s, bufferDequeue::getTimeOfLastRecord));
+
     return streams.stream()
-        .sorted(Comparator.comparing((StreamDescriptor s) -> bufferDequeue.getQueueSizeBytes(s).orElseThrow(), Comparator.reverseOrder())
```

Comment:
Our theory is prior to this implementation, these values could change mid-comparison, and thus cause the `java.lang.IllegalArgumentException: Comparison method violates its general contract!` we see.

Reply:
this theory makes sense to me! fun times with 🧵🧵🧵